agent/agent.go

```diff
@@ -3700,6 +3737,24 @@ func (a *Agent) ReloadConfig() error {
 	// breaking some existing behavior.
 	newCfg.NodeID = a.config.NodeID
 
+	//if auto reload is enabled, make sure we have the right certs file watched.
+	if autoReload {
+		for _, f := range []struct {
+			oldCfg string
+			newCfg string
+		}{{a.config.KeyFile, newCfg.KeyFile}, {a.config.KeyFile, newCfg.KeyFile}} {
```

Comment:
I am very confused because both of these slice tuples are identical.

Reply:
yes your right, copy pasta mistake and I need to add a test case!!